include/posix/pthread.h

```diff
@@ -157,42 +159,61 @@ static inline int pthread_condattr_destroy(pthread_condattr_t *att)
 		.sem = &name##_psem,		\
 	}
 
-/**
- * @brief POSIX threading compatibility API
+/*
+ *  Mutex attributes - type
+ *
+ *  PTHREAD_MUTEX_NORMAL: Owner of mutex cannot relock it. Attempting
+ *      to relock will cause deadlock.
+ *  PTHREAD_MUTEX_RECURSIVE: Owner can relock the mutex.
+ *  PTHREAD_MUTEX_ERRORCHECK: If owner attempts to relock the mutex, an
+ *      error is returned.
  *
- * See IEEE 1003.1
  */
-static inline int pthread_mutex_init(pthread_mutex_t *m,
-				     const pthread_mutexattr_t *att)
-{
-	ARG_UNUSED(att);
+#define PTHREAD_MUTEX_NORMAL        0
+#define PTHREAD_MUTEX_RECURSIVE     1
+#define PTHREAD_MUTEX_ERRORCHECK    2
+#define PTHREAD_MUTEX_DEFAULT       PTHREAD_MUTEX_NORMAL
 
-	k_sem_init(m->sem, 1, 1);
+#define MUTEX_MAX_REC_LOCK 32767
 
-	return 0;
-}
+/* This timeout is not standard value. Set for convenience. */
+#define MEM_TIMEOUT 100
+
+/*
+ *  Mutex attributes - protocol
+ *
+ *  PTHREAD_PRIO_NONE: Ownership of mutex does not affect priority.
+ *  PTHREAD_PRIO_INHERIT: Owner's priority is boosted to the priority of
+ *      highest priority thread blocked on the mutex.
+ *  PTHREAD_PRIO_PROTECT:  Mutex has a priority ceiling.  The owner's
+ *      priority is boosted to the highest priority ceiling of all mutexes
+ *      owned (regardless of whether or not other threads are blocked on
+ *      any of these mutexes).
+ */
+#define PTHREAD_PRIO_NONE           0
+#define PTHREAD_PRIO_INHERIT        1
```

Comment:
Code might test with `#ifdef PTHREAD_PRIO_INHERIT` if a certain priority level is supported by the system prior to using it.  Please use `__DEPRECATED_MACRO` or something similar to note that this macro shouldn't be used as it's not supported.

Reply:
Done